Rubberduck.Parsing/Annotations/ParseTreeAnnotation.cs

```diff
@@ -0,0 +1,81 @@
+﻿using Rubberduck.Parsing.Grammar;
+using Rubberduck.VBEditor;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Rubberduck.Parsing.Annotations
+{
+    public class ParseTreeAnnotation
+    {
+        public const string ANNOTATION_MARKER = "@";
+
+        private readonly Lazy<int?> _annotatedLine;
+        
+        internal ParseTreeAnnotation(IAnnotation annotation, QualifiedSelection qualifiedSelection, VBAParser.AnnotationContext context)
+        {
+            QualifiedSelection = qualifiedSelection;
+            Context = context;
+            Annotation = annotation;
+            _annotatedLine = new Lazy<int?>(GetAnnotatedLine);
+            AnnotationArguments = AnnotationParametersFromContext(Context);
+        }
+
+        // FIXME annotation constructor for unit-testing purposes alone!
+        internal ParseTreeAnnotation(IAnnotation annotation, QualifiedSelection qualifiedSelection, IEnumerable<string> arguments)
```

Comment:
I think that shows that the class isn't a POCO and thus needs an interface so we can mock it easily in our unit tests without having to have the hacky ctor and cross our fingers nobody calls that ctor.

Reply:
interface extracted, test code using the constructor converted to mocks